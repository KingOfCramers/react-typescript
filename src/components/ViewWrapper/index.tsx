import React from "react";
import Footer from "../Footer";
import Header from "../Header";

interface ViewWrapperProps {
  View: React.FC;
}

const ViewWrapper = ({ View }: ViewWrapperProps) => {
  return (
    <>
      <Header />
      <View />
      <Footer />
    </>
  );
};

export default ViewWrapper;
