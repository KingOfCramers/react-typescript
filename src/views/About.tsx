import React from "react";

interface AboutProps {}

const About = (props: AboutProps) => {
  return <div>About Page</div>;
};

export default About;
