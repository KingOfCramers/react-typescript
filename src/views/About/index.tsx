import React from "react";

interface AboutProps {}

const About = (props: AboutProps) => {
  return <div>This application was designed</div>;
};

export default About;
