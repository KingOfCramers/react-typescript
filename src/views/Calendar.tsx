import React from "react";

interface CalendarProps {}

const Calendar = (props: CalendarProps) => {
  return <div>Calendar View</div>;
};

export default Calendar;
