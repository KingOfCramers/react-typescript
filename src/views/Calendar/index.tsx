import React from "react";

interface CalendarProps {}

const Calendar = (props: CalendarProps) => {
  return (
    <div>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Commodi
      recusandae adipisci, ducimus consequatur iusto ipsa repellat voluptatum
      laborum! Minima possimus rerum dicta quia, unde saepe explicabo eum dolor
      excepturi ut aliquam reiciendis? Earum, incidunt recusandae possimus eius
      animi voluptatibus perspiciatis quasi cumque sunt optio veniam vitae nisi,
      culpa ipsam repellendus tempora sint officiis illo. Provident doloribus
      corrupti ut eaque, nihil magnam dicta quas ullam recusandae natus ab a
      voluptas sunt quasi distinctio earum accusamus maiores aliquam mollitia
      nobis autem, iusto, nulla maxime! Ut id aspernatur, cum perspiciatis quos
      earum laboriosam ab odio optio blanditiis fuga nam laudantium dolore hic
      adipisci sapiente similique ducimus asperiores labore nemo doloribus fugit
      voluptatum eveniet. Perspiciatis sapiente vero ratione recusandae. Sunt
      quod labore alias impedit at iste adipisci est explicabo dolore cupiditate
      molestias doloribus, neque nisi quibusdam. Recusandae dolores est
      excepturi velit. Ea, odio delectus minima repudiandae maiores assumenda
      voluptatem aut veniam ipsum ipsa tempore! Autem, reiciendis cum? Repellat
      sed enim, quibusdam tenetur ad necessitatibus a reprehenderit molestias
      voluptas laboriosam officia? Ad laudantium voluptate facere quibusdam esse
      nostrum hic eligendi saepe nesciunt ab deserunt, sunt assumenda veniam
      ullam magnam est nisi cumque dolorem! Perferendis dolorum voluptatum
      eveniet soluta sed repudiandae laudantium quod aperiam itaque. Sequi
      reiciendis consequatur molestiae delectus totam? Deserunt officiis
      eveniet, possimus, velit illo est qui eligendi dolorem cupiditate, ipsa
      exercitationem magnam neque. Nesciunt ea deserunt porro accusamus
      perspiciatis possimus id natus error iure commodi debitis maiores harum
      odio quos, consequuntur est exercitationem autem quaerat et! Tempore
      aliquid repellat dolorem dolores dicta eum quos sapiente? Facilis
      aspernatur illum eius ratione quos vitae cum impedit minima rerum,
      architecto at deserunt est fugiat, veniam sequi id. Maiores officia
      explicabo consectetur? Quas dolore, nesciunt facilis recusandae dicta
      libero qui, mollitia, enim tempore fuga quo reiciendis quae a doloribus
      magnam voluptatibus reprehenderit. Ipsum, at minima recusandae saepe,
      earum temporibus architecto necessitatibus quo suscipit exercitationem
      facilis illo quos explicabo iste ducimus quisquam odit. Non tempore,
      aliquam esse, vitae aliquid sed, autem similique harum et hic dolores
      consequuntur quidem mollitia debitis blanditiis. Sunt ipsam aut eius
      recusandae ducimus nam itaque nisi officiis consequuntur debitis
      voluptatem deserunt corporis sed aliquid ex nostrum quae quas asperiores
      nobis, autem maxime. Provident eligendi deserunt suscipit neque aut
      ducimus ad iste repellendus ratione amet facilis numquam vel aliquid
      perspiciatis cumque itaque laborum temporibus accusantium ex, officiis
      delectus ipsa nulla praesentium! Amet eaque a laborum recusandae tempore
      enim atque quisquam iusto, exercitationem molestiae vero nisi et aliquid
      sed distinctio magni, dolores illum delectus deserunt consequatur maiores!
      Voluptatibus, tempore similique reiciendis vitae temporibus maiores
      dolorum laudantium corporis praesentium a iusto libero molestias impedit
      sequi maxime sit porro culpa reprehenderit quaerat error officia ad
      dignissimos blanditiis? Est possimus natus tempore. Amet aperiam pariatur
      repudiandae deserunt vero ducimus vitae quibusdam, facere optio
      reprehenderit voluptatibus corporis quam expedita molestiae cum distinctio
      nostrum sint molestias numquam eaque aspernatur a beatae quisquam. Tempora
      dignissimos illum quisquam? Est corporis, architecto nam maiores maxime
      earum molestiae aliquid exercitationem inventore asperiores necessitatibus
      iste recusandae similique sequi cum possimus culpa obcaecati. Aspernatur
      aliquid cumque aliquam, distinctio officia, eos impedit cum nostrum,
      exercitationem vel quibusdam ipsam! Laborum excepturi officia ipsam rerum
      sint a consectetur, pariatur illum magni laboriosam deserunt! Placeat
      cupiditate quaerat corporis similique eos ab, quae reprehenderit, quia
      adipisci debitis praesentium error earum quos facilis incidunt
      necessitatibus eius sequi sapiente ipsam nam consequuntur? Soluta officiis
      ad molestias harum dolor reiciendis molestiae laboriosam provident libero
      nihil, reprehenderit, voluptates accusamus suscipit odit quod in deleniti
      incidunt quo dolorum aperiam quos doloribus expedita. Ut reprehenderit ea,
      tenetur adipisci ex facilis aliquid nam vero aut tempore accusantium non
      odit beatae itaque sed incidunt enim sapiente voluptatum unde praesentium
      dolores suscipit, libero, numquam provident! At dolor cum necessitatibus
      tempore magnam error officiis et porro, nobis totam quod quidem eius minus
      laboriosam commodi sit ex! Eius saepe officia illum aperiam consectetur
      itaque incidunt quasi voluptatem, reprehenderit magni, quaerat consequatur
      laboriosam! Alias, doloribus dicta excepturi distinctio perspiciatis
      reiciendis a non laudantium animi laborum debitis sint dolores culpa
      voluptates eos, veritatis voluptatum quia et? Ut quaerat, rerum expedita
      quos inventore, dignissimos libero delectus labore distinctio voluptatem
      eius tenetur sit quasi incidunt fuga illo recusandae. Hic tempora nostrum
      impedit vero dolore amet tempore delectus, qui excepturi at minus veniam
      eveniet odit, aspernatur quidem odio in quam fugiat, debitis mollitia
      obcaecati. Modi officiis eligendi magni? Natus saepe eum provident sed
      suscipit molestiae excepturi aut corporis necessitatibus, aliquid neque
      illum nam vero eligendi dolore aspernatur ipsum reprehenderit consequatur
      amet id, sit mollitia sequi totam? Eveniet quo, quam quasi iusto
      perferendis ipsum sunt, repudiandae modi sit error soluta enim facilis
      doloribus eius aliquid porro quisquam ab ad neque. Distinctio provident
      laudantium aperiam beatae officiis. Aut ducimus dignissimos harum iure
      laudantium et, itaque veritatis a maiores nostrum, placeat velit! Fugiat
      perspiciatis molestiae, commodi et eum molestias aspernatur quod dolorem
      alias id saepe recusandae, possimus illum blanditiis nisi? Rem ullam,
      commodi eum molestiae minima ipsum consequatur enim reprehenderit
      praesentium earum voluptatibus et placeat quis soluta deserunt tempora,
      voluptatem minus corporis consequuntur recusandae inventore? Pariatur non,
      ipsa tempora suscipit quos iste enim dolor aliquid debitis, quae, autem at
      soluta vel voluptatum vero velit adipisci tenetur aperiam vitae eius
      ducimus. Pariatur dignissimos iste, vel similique veritatis itaque, totam
      ad possimus accusantium ab eaque minus odit? Dolore aut asperiores
      deleniti eveniet sequi, excepturi repellat error rerum in consequuntur
      tempore nihil hic. Deleniti iure aperiam provident perferendis sapiente
      nisi laborum aliquam asperiores dolorem, pariatur ratione, corrupti
      consequuntur accusantium facilis cum optio odit beatae delectus dolore
      nostrum eum? Facilis similique nisi incidunt maiores neque ratione libero
      assumenda voluptatibus nemo expedita, nam, quam perspiciatis debitis,
      dolor aut asperiores ipsa amet sunt dolore? Modi incidunt eius impedit sit
      quod quis facilis, officiis beatae ut sint rem repudiandae et. Accusamus,
      laboriosam! Omnis voluptatibus alias consectetur sit laudantium neque
      molestias eaque, voluptates, dignissimos quod quaerat possimus adipisci in
      rem, pariatur non asperiores. Corporis reiciendis molestias sapiente quae
      mollitia minus laudantium odit voluptates veritatis sit possimus beatae
      dolores in quod, odio, praesentium obcaecati sed culpa maxime optio?
      Commodi ipsa autem pariatur in cum distinctio voluptate dolor quos
      explicabo quisquam?
    </div>
  );
};

export default Calendar;
