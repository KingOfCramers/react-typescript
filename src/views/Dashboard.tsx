import React from "react";

interface DashboardProps {}

const Dashboard = (props: DashboardProps) => {
  return (
    <div>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Sequi,
      repudiandae! Dolores, voluptate nemo voluptatibus provident consequuntur
      repudiandae nihil nisi nulla soluta, quaerat odit quidem alias porro.
      Iusto, dolorum. Iure dignissimos provident dolorum numquam sit. A nemo
      vero ducimus aperiam id eos sunt ratione assumenda fuga, magnam saepe esse
      nulla in asperiores inventore, maiores, cum libero dolorum exercitationem
      excepturi. Unde libero, dolor sequi placeat, repellendus beatae est
      pariatur quibusdam officia, error consequatur eum voluptate repellat eaque
      aperiam. Fugit, dicta. Incidunt reiciendis dolor veritatis quasi,
      voluptatum amet. Ratione ipsum beatae voluptatibus ullam praesentium,
      officiis, dolorum repellendus obcaecati eum nihil pariatur eos. Vitae, ea
      esse, tempora officia temporibus natus similique, maxime quaerat rerum
      corporis ipsum consectetur dolorum animi fuga odio sed perferendis
      accusantium omnis maiores sint tempore itaque deleniti expedita voluptas.
      Maxime illo in vel sequi autem, blanditiis placeat perferendis fuga unde
      quidem magni voluptatibus voluptates consequatur eligendi modi totam
      ratione. Architecto ducimus quo odit totam maiores, temporibus nisi
      reiciendis officiis repudiandae ipsum assumenda harum officia aliquid
      laboriosam. Adipisci voluptates neque eligendi molestiae! Ad ratione
      adipisci minus maxime, dolor mollitia et autem dolore alias omnis
      architecto fugiat atque esse blanditiis. Maiores incidunt reprehenderit
      eos veniam blanditiis ad enim. Pariatur porro voluptas ratione repellendus
      maiores commodi laborum iste explicabo culpa hic, unde, similique, modi
      totam ex tempore quo. Corrupti rem aut blanditiis unde! Saepe, sequi
      obcaecati. Nesciunt rerum excepturi enim natus quos corporis nihil
      mollitia fugit, pariatur nemo quibusdam unde, deserunt cum sapiente et
      doloremque assumenda cumque, atque ratione dolores temporibus. Illum
      provident obcaecati sapiente accusantium harum animi reiciendis rerum
      aperiam delectus, laboriosam tempore iste consequuntur, ad incidunt qui
      possimus? Porro blanditiis quia eveniet possimus unde hic illum impedit
      explicabo, corrupti officia voluptate, quae mollitia molestias
      repudiandae. Quidem in ea, sed, consequatur amet deserunt, sint aperiam
      ducimus necessitatibus beatae velit qui iusto repellat eum nihil incidunt
      non dicta blanditiis ad id tenetur accusamus quas suscipit doloribus!
      Esse, sed. Ipsam, labore. Repellendus ipsam explicabo, deleniti dolor
      natus illum. Pariatur doloremque reiciendis cupiditate impedit quae,
      culpa, beatae error aliquid sequi sed, tempora magni necessitatibus
      reprehenderit! Nostrum, vitae necessitatibus sequi ad molestiae ex vero
      ipsa itaque assumenda, facilis optio voluptatem, voluptatibus blanditiis
      ipsum error possimus similique quasi eum placeat aliquam minima! Iusto
      sapiente officia harum ab sequi consectetur illum explicabo delectus
      possimus aut aliquam eius id necessitatibus consequatur dolor inventore
      dolore eligendi qui, praesentium magnam unde cumque placeat obcaecati
      beatae. Omnis inventore natus dignissimos harum excepturi, enim dolores
      eos eum libero esse mollitia odit officia sint eius ullam accusantium quae
      consequuntur molestias unde deserunt quidem laborum illum exercitationem
      voluptates? Distinctio voluptatum molestiae veritatis assumenda omnis. Sed
      minus eos a voluptas? Soluta sed quaerat unde. Minus dolorum illum, unde
      vero, debitis saepe id fugit expedita libero, optio ratione fugiat magni
      iusto illo quia accusantium? Obcaecati, accusamus neque. Quod provident
      facilis accusantium doloremque nostrum repellat voluptate nihil at iusto
      architecto sit perspiciatis suscipit asperiores harum odio nulla
      voluptates magnam aspernatur, modi ut aperiam, soluta excepturi. Esse
      repellat dolor reprehenderit. Officia placeat illo autem repellendus,
      consequatur perferendis? Odit quae aspernatur omnis! Sint, earum totam
      pariatur architecto molestias consequuntur possimus repellendus tempora
      mollitia quis hic magnam, facere officia quae explicabo. Inventore nemo
      doloremque nulla quisquam quasi quis voluptas! Aspernatur pariatur ut
      nobis, rerum, nostrum quibusdam officia praesentium cum quas aliquam eaque
      magni quod ea quisquam magnam, sequi molestias optio! Ratione, accusamus
      voluptas. Quaerat, repellat officiis ab unde aliquam corrupti minus
      commodi maxime omnis quia, optio eveniet suscipit magni amet quibusdam
      quidem. Excepturi rem, facilis culpa tempora vero qui quisquam, facere
      sequi error, alias nihil eveniet. Laboriosam nisi tenetur consectetur,
      architecto culpa accusantium doloremque possimus nihil distinctio
      similique totam quidem harum laudantium dolor ratione sapiente odio optio
      doloribus ducimus quaerat ea. Omnis veritatis aut cumque consequatur,
      illum velit fugiat quam officia nobis atque excepturi magni animi quia
      numquam impedit deserunt ex sed enim facere. Eligendi, ratione temporibus?
      Tenetur doloremque quam doloribus iusto perspiciatis? Quidem, magni
      repellat. Inventore repudiandae omnis eveniet possimus hic doloremque
      nihil vitae vel, error iste velit facilis magni! Veniam perspiciatis porro
      sequi. Enim facilis quod laborum dolore voluptas odit ab ea nobis quia
      exercitationem sequi quo, doloribus odio aspernatur a ex praesentium autem
      voluptates tempore! Reprehenderit quibusdam illo, dicta ex quam nihil
      aliquam deleniti laborum laudantium architecto ab assumenda tenetur
      soluta, distinctio itaque quaerat aperiam commodi ad atque. Ipsa explicabo
      quidem ut voluptatum labore nobis illo doloremque, tempora iure in laborum
      aliquam, doloribus vel? Quibusdam, eligendi porro? Itaque consequuntur
      amet beatae molestias obcaecati sequi. Vel perferendis quaerat quidem
      laudantium, sit ullam nemo accusantium amet labore fuga quasi ad odio
      adipisci incidunt maiores magnam earum, maxime non nisi. Aliquam,
      asperiores corrupti? Fugit voluptatum deleniti quidem, veniam nobis sint
      quis distinctio eos eum quod quisquam illo, ipsum ratione inventore ullam
      reiciendis dolorem esse alias dicta. Corrupti accusamus vero libero.
      Aperiam est libero, totam hic obcaecati distinctio ab cumque voluptatem
      molestias ex aliquid maxime cupiditate laborum alias deserunt quis
      dolorem, tempora suscipit. Porro, consequatur? Ratione a, accusamus sit
      quo in maiores ab fugit exercitationem corporis? Delectus rerum aperiam
      illum, molestias quaerat quibusdam iusto asperiores eos ut accusantium,
      fugit quo natus pariatur sapiente? Omnis voluptates exercitationem, sint
      sed ducimus pariatur accusamus nam, magnam facere modi nobis. Ullam ex ea
      et rerum cumque perspiciatis quo sapiente voluptate eveniet! Voluptatum,
      odit! Facilis quia sunt eaque ab doloremque atque nostrum dignissimos
      suscipit saepe, impedit blanditiis laborum asperiores reiciendis
      repellendus quasi excepturi necessitatibus numquam laboriosam quaerat quo
      minus eius maxime natus sint. Non nemo officiis, doloremque eveniet quae
      maiores sint! Nulla consectetur quo facilis eaque corrupti distinctio
      tempore quasi labore quidem harum tenetur iste sunt repellat ducimus
      laudantium, quam animi ipsum officiis inventore vel dolorem consequatur.
      Fuga libero optio quae cumque eaque sequi minus quia cum ipsum natus,
      laboriosam architecto perferendis. Aspernatur corporis iste ipsum nobis
      iusto laboriosam numquam odit a cum, fuga placeat ut autem omnis hic enim
      dolor porro reprehenderit doloremque aperiam eveniet quis excepturi veniam
      deleniti. Perspiciatis quas nostrum eligendi ut repellat odit nemo
      necessitatibus corrupti ex placeat, molestiae officia id consectetur
      voluptatibus tempore expedita, doloribus dolore quibusdam maiores, labore
      asperiores suscipit ratione. Ipsum?
    </div>
  );
};

export default Dashboard;
