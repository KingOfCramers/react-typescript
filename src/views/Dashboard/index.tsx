import React, { ReactElement } from "react";
import { useQuery } from "@apollo/react-hooks";
import { useQueryParam, StringParam } from "use-query-params";
import { GET_BOOKS } from "../../graphql/queries";

import TextBox from "../../components/TextBox";

interface DashboardProps {}

const Dashboard = (props: DashboardProps): ReactElement | null => {
  const [filter, setFilter] = useQueryParam("filter", StringParam);

  const { loading, error, data, fetchMore } = useQuery(GET_BOOKS);
  if (loading) return <div>Loading...</div>;
  if (error) return <div>`Error! ${error.message}`</div>;

  const handleOnChange = (val: string): void => {
    setFilter(val);
  };

  return (
    <div>
      <TextBox
        placeholder="Search..."
        word={filter || ""}
        callBack={handleOnChange}
      />
      <p>{filter}</p>
    </div>
  );
};

export default Dashboard;
