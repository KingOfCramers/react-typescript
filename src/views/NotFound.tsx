import React from "react";

interface NotFoundProps {}

const NotFound = (props: NotFoundProps) => {
  return <div>Not found view</div>;
};

export default NotFound;
