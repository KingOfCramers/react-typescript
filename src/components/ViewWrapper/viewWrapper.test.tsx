import React from "react";
import { shallow } from "enzyme";
import ViewWrapper from "./index";

import Dashboard from "../../views/Dashboard";
import Calendar from "../../views/Calendar";
import About from "../../views/About";
import NotFound from "../../views/NotFound";

test("Renders the dashboard view", () => {
  const component = shallow(<ViewWrapper View={Dashboard} />);
  expect(component).toMatchSnapshot();
});
test("Renders the calendar view", () => {
  const component = shallow(<ViewWrapper View={Calendar} />);
  expect(component).toMatchSnapshot();
});
test("Renders the about view", () => {
  const component = shallow(<ViewWrapper View={About} />);
  expect(component).toMatchSnapshot();
});
test("Renders the notfound view", () => {
  const component = shallow(<ViewWrapper View={NotFound} />);
  expect(component).toMatchSnapshot();
});
