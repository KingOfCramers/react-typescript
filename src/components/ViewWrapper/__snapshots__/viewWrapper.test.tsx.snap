// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the about view 1`] = `
<Fragment>
  <Memo(Header) />
  <About />
  <Footer />
</Fragment>
`;

exports[`Renders the calendar view 1`] = `
<Fragment>
  <Memo(Header) />
  <Calendar />
  <Footer />
</Fragment>
`;

exports[`Renders the dashboard view 1`] = `
<Fragment>
  <Memo(Header) />
  <Dashboard />
  <Footer />
</Fragment>
`;

exports[`Renders the notfound view 1`] = `
<Fragment>
  <Memo(Header) />
  <NotFound />
  <Footer />
</Fragment>
`;
